### PR TITLE
Add email headers

### LINKED ISSUE #15926: Emails need metadata in the headers (e.g. repo, notice kind, issue state)
Emails should have some basic metadata in the headers to convey this information:

* forge identifier
* repository identifier
* notice kind (mention, new issue, issue comment, commit alert, [new assignment](https://github.com/go-gitea/gitea/issues/2141), etc)
* issue number & state (if it's an issue, state whether the issue is open, closed, etc)
* unsubscribe link (a unique link to enable recipients to unwatch a topic without having to login)

Github does this relatively well and I suggest modeling somewhat after Github.  E.g.
```
X-GitHub-Sender: Alice
X-GitHub-Recipient: Bob
X-GitHub-Reason: mention
List-ID: guardianproject/haven <haven.guardianproject.github.com>
List-Archive: https://github.com/guardianproject/haven
List-Unsubscribe: <mailto:unsub+IKEF_yadayada_CVQAQ@reply.github.com>
X-GitHub-Recipient-Address: bob@bobsemaildomain.tld
```

## Rationale:

It's useful for recipients to have headers by which to filter their inbound email.  Advanced users write filters for tools like Procmail and Sieve, and they generally do it in a generic way.  Instead of manually creating an inbox like "codeberg_repo_A" and then manually writing a filter for each repo they work on, advanced users write just one filter that parses the metadata from the headers and auto generates inboxes on-the-fly.  This is particularly important for gitea because unlike Github there are many Gitea forges and projects are scattered.

## forge-custom headers insufficient:

Someone raised an issue that was vaguely similar, and the [solution](https://github.com/go-gitea/gitea/issues/5083#issuecomment-688486945) was insufficient.  They made it possible for every forge to add their own [custom headers](https://github.com/go-gitea/gitea/issues/6008).  That's useful for some things but lousy for the metadata above.  It's a bad idea for every Gitea forge to customize the same basic info in different ways because it requires users to create a different mail filter for every gitea instance.

## to-from-subject-body <= bad places

The [suggestion](https://github.com/go-gitea/gitea/issues/5083) to use "To", "From", "Subject", and email bodies is bad -- far from ideal.  Some mail servers and processors (particularly forwarding services and tools like spamassassin) will amend/corrupt/destroy/munge/botch the following headers in various bizarre ways that makes these specific headers unsuitable for mail filtering:

* To
* From
* Subject

One mail forwarding service completely removes all traces of the original "To" address which makes life difficult, so paradoxically even the `X-GitHub-Recipient-Address` header is quite useful as well. The body is also a bad place because procmail code gets pretty ugly and error prone when it must look at the body, which is also slower.

---
gitea version: v1.14.2
source: https://codeberg.org/Codeberg/Community/issues/450

### PR DESCRIPTION
**Pull request by KN4CK3R**
fixes #15926
related #13585, #10808, #13283

Added some custom headers to issue emails to support better filters. See #15926 for details.

Like in webhooks I have added `X-GitHub-` and `X-GitLab-` headers too but want your opinion if we want to send them or not. The `...-Reason` header may not match what they are sending.

Part of a test mail (sorted):
```
In-Reply-To: <KN4CK3R/TestRepo/issues/1@localhost>
List-Archive: <http://localhost:3000/KN4CK3R/TestRepo>
List-ID: KN4CK3R/TestRepo <TestRepo.KN4CK3R.localhost>
References: <KN4CK3R/TestRepo/issues/1@localhost>
Subject: Re: [KN4CK3R/TestRepo] Test (#1)
To: kn4ck3r@gitea.test
X-Auto-Response-Suppress: All
X-GitHub-Reason: issue
X-GitHub-Recipient-Address: kn4ck3r@gitea.test
X-GitHub-Recipient: KN4CK3R
X-GitHub-Sender: KN4CK3R2
X-GitLab-Issue-IID: 1
X-GitLab-NotificationReason: issue
X-GitLab-Project-Path: KN4CK3R/TestRepo
X-GitLab-Project: TestRepo
X-Gitea-Issue-ID: 1
X-Gitea-Issue-Link: http://localhost:3000/KN4CK3R/TestRepo/issues/1
X-Gitea-Reason: issue
X-Gitea-Recipient-Address: kn4ck3r@gitea.test
X-Gitea-Recipient: KN4CK3R
X-Gitea-Repository-Link: http://localhost:3000/KN4CK3R/TestRepo
X-Gitea-Repository-Path: KN4CK3R/TestRepo
X-Gitea-Repository: TestRepo
X-Gitea-Sender: KN4CK3R2
```

### CODE REVIEW

**Comment by silverwind** [comment]
> Like in webhooks I have added X-GitHub- and X-GitLab- headers too but want your opinion if we want to send them or not.

Fine with me, we should do it consistently.

File: services/mailer/mail.go [hunk 1]
```diff
@@ -277,12 +278,51 @@ func composeIssueCommentMessages(ctx *mailCommentContext, lang string, tos []str
...
+
+		"X-GitHub-Reason":            reason,
+		"X-GitHub-Sender":            ctx.Doer.DisplayName(),
+		"X-GitHub-Recipient":         recipient.DisplayName(),
+		"X-GitHub-Recipient-Address": recipient.Email,
+
+		"X-GitLab-NotificationReason": reason,
+		"X-GitLab-Project":            repo.Name,
+		"X-GitLab-Project-Path":       repo.FullName(),
+		"X-GitLab-Issue-IID":          strconv.FormatInt(ctx.Issue.Index, 10),
```
Comment (silverwind) [inline services/mailer/mail.go:322]: `IID` is no typo, right?

Comment (KN4CK3R) [reply services/mailer/mail.go:322]: No typo. They use `IID` for the repo specific count (Issue 1) and `ID` for the internal/database id.

**Review by silverwind — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Comment by camlafit** [comment]
Thanks a lot :)

**Pull request merged**